opsdroid/loader.py

```diff
@@ -349,12 +350,21 @@ def include_constructor(loader, node):
         yaml.SafeLoader.add_constructor("!include", include_constructor)
 
         try:
+            schema_path = os.path.abspath("opsdroid/configuration/schema.yaml")
+            schema = yamale.make_schema(schema_path)
+            data = yamale.make_data(config_path)
+            yamale.validate(schema, data)
             with open(config_path, "r") as stream:
                 _LOGGER.info(_("Loaded config from %s."), config_path)
                 return yaml.safe_load(stream)
+        except ValueError as error:
+            _LOGGER.critical(error)
+            sys.exit(1)
```

Comment:
I would prefer it if you could call [`opsdroid.critical`](https://github.com/opsdroid/opsdroid/blob/8b252588aeb2f7bfcfda01845b5c9c985c2b19a8/opsdroid/core.py#L108) here and pass it a friendly error message to log along with the one from `yamale`.

This is just so we can handle criticals consistently and if we need to update that in the future we can do it in one place. The loader should have a pointer to opsdroid as an attribute.

Reply:
load_config_file - class method, it is not possible to call the opsdroid.critical method

Reply:
Ok sure. Perhaps this highlights that `opsdroid.critical` should also be a classmethod. This would require some change to the exit logic so let's leave it for now.